include/deal.II/lac/solver_gmres.h

```diff
@@ -2079,6 +2332,63 @@ void SolverFGMRES<VectorType>::solve(const MatrixType         &A,
                 SolverControl::NoConvergence(accumulated_iterations, res));
 }
 
+
+
+//----------------------------------------------------------------------//
+
+
+
+template <typename VectorType>
+DEAL_II_CXX20_REQUIRES(concepts::is_vector_space_vector<VectorType>)
+SolverFGMRES<VectorType>::SolverFGMRES(SolverControl            &cn,
+                                       VectorMemory<VectorType> &mem,
+                                       const AdditionalData     &data)
+  : SolverMPGMRES<VectorType>(
+      cn,
+      mem,
+      typename SolverMPGMRES<VectorType>::AdditionalData{
+        data.max_basis_size,
+        true,
+        data.orthogonalization_strategy})
+{
+  this->indexing_strategy = SolverMPGMRES<VectorType>::IndexingStrategy::fgmres;
```

Comment:
This is awkward. I *think* what you are doing is set a flag in the base class. I think a better design choice would be to add this as an argument to `solve_internal` so that you don't need the `protected` member variable in the base class.